src/planner/binder.cpp

```diff
@@ -306,13 +306,14 @@ void Binder::AddCorrelatedColumn(const CorrelatedColumnInfo &info) {
 	}
 }
 
-bool Binder::HasMatchingBinding(const string &table_name, const string &column_name, string &error_message) {
+bool Binder::HasMatchingBinding(const string &table_name, const string &column_name, string &error_message,
+                                TableColumnType type) {
 	string empty_schema;
-	return HasMatchingBinding(empty_schema, table_name, column_name, error_message);
+	return HasMatchingBinding(empty_schema, table_name, column_name, error_message, type);
 }
 
 bool Binder::HasMatchingBinding(const string &schema_name, const string &table_name, const string &column_name,
-                                string &error_message) {
+                                string &error_message, TableColumnType type) {
```

Comment:
Is it required for the `HasMatchingBinding` to have this parameter? Can we try to localize this in the `TableBinding`?